pkg/kn/commands/service/service_update.go

```diff
@@ -53,24 +55,33 @@ func NewServiceUpdateCommand(p *commands.KnParams) *cobra.Command {
 				return err
 			}
 
-			service, err := client.Services(namespace).Get(args[0], v1.GetOptions{})
-			if err != nil {
-				return err
-			}
-			service = service.DeepCopy()
+			var retries = 0
+			for true {
+				service, err := client.Services(namespace).Get(args[0], v1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				service = service.DeepCopy()
```

Comment:
i dont think we need DeepCopy

Reply:
Not introduced in this PR, but happy to remove if safe. it was just my general understanding to copy objects before working on them to not compromise shared caches and other data structures which hold a reference.

Reply:
I'm with rhuss.